File: Python/src/Problem002.py
__author__ = 'Joel'


def even_fibonacci_below(limit):
    a = 1
    b = 2
    total = 2 # Starting at 2 because the below excludes the first two values of the sequence.
    pos = 0
    while b < limit:
        a, b = b, a + b
        if pos == 2:
            total += b
            pos = 0
        else:
            pos += 1
    return total


print(even_fibonacci_below(4000000))

File: Java/src/solutions/Problem002.java
package solutions;

import framework.Problem;
import framework.ProblemInfo;

@ProblemInfo(
        description = "Even Fibonacci numbers",
        number = 2
)
public class Problem002 extends Problem {
    @Override
    public void solve() {
        int a = 1;
        int b = 2;
        int total = 2; //Starting at 2 because the below excludes the first two values of the sequence.
        while (b < 4000000) {
            int c = a + b;
            a = b;
            b = c;
            if (c % 2 == 0) {
                total += c;
            }
        }
        System.out.println(total);
    }
}